gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/TeradataExtractor.java

```diff
@@ -188,12 +192,15 @@ public TeradataExtractor(WorkUnitState workUnitState) {
 
   @Override
   public String getConnectionUrl() {
+    String urlPrefix = "jdbc:teradata://"; 
     String host = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_HOST_NAME);
+    checkArgument(!isNullOrEmpty(host), "Connectionn host cannot be null or empty at %s", ConfigurationKeys.SOURCE_CONN_HOST_NAME);
+    
     String port = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_PORT,"1025");
     String database = this.workUnit.getProp(ConfigurationKeys.SOURCE_QUERYBASED_SCHEMA);
-    String url = "jdbc:teradata://" + host.trim() + "/DATABASE=" + database.trim() + ",DBS_PORT=" + port.trim() ;
-    
-    return url;
+    String defaultUrl = urlPrefix + host.trim() + "/DATABASE=" + database.trim() + ",DBS_PORT=" + port.trim() ;
```

Comment:
I think it's worth to add the `TYPE=FASTEXPORT` connection parameter by default.